android/app/src/main/java/com/google/codeu/chatme/view/adapter/UserListAdapter.java

```diff
@@ -0,0 +1,76 @@
+package com.google.codeu.chatme.view.adapter;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.google.codeu.chatme.R;
+import com.google.codeu.chatme.model.User;
+import com.google.codeu.chatme.presenter.UserPresenter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A {@link android.support.v7.widget.RecyclerView.Adapter} to bind the list of users
+ * to the recyclerview in {@link com.google.codeu.chatme.view.tabs.UsersFragment}
+ */
+public class UserListAdapter extends RecyclerView.Adapter<UserListAdapter.ViewHolder>
+        implements UserListAdapterView {
+
+    private List<User> users = new ArrayList<>();
+
+    private UserPresenter presenter;
+
+    public UserListAdapter() {
+        this.presenter = new UserPresenter(this);
+    }
+
+
+    @Override
+    public UserListAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.user_list_item, parent, false);
+        return new ViewHolder(v);
+    }
+
+    public void loadUsers() {
+        this.presenter.loadUsers();
+    }
+
+
+    @Override
+    public void onBindViewHolder(UserListAdapter.ViewHolder holder, int position) {
+        final User user = users.get(position);
+        holder.tvName.setText(user.getName());
```

Comment:
name has been changed to`fullName` and `username` in the  `User` model. merge the latest version of `dev` to this branch and make the necessary changes. I would suggest attempting to grab each user's Full Name first and, if empty, grab their username instead.